src/NumericalAlgorithms/Interpolation/LinearInterpolationND.hpp

```diff
@@ -0,0 +1,374 @@
+    extrapolate_high[whichDimension] = value;
+  }
+
+  inline void extrapolateLow(size_t whichDimension, bool value) {
+    extrapolate_low[whichDimension] = value;
+  }
+
+  inline double const& xmin(size_t whichDimension) {
+    return x[whichDimension][0];
+  };
+
+  inline double const& xmax(size_t whichDimension) {
+    return x[whichDimension][numberOfPoints[whichDimension] - 1];
+  };
+
+  template <
+      size_t... whichVariables, typename... Xt,
+      Requires<(std::is_convertible_v<typename std::remove_cv_t<Xt>, double> &&
+                ...)> = nullptr>
+  inline std::array<double, sizeof...(whichVariables)> interpolate(
+      Xt const&... xin) const {
+    static_assert(
+        sizeof...(Xt) == Dimension,
+        "You need to provide the correct number of interpolation points");
+    static_assert(sizeof...(whichVariables) <= NumberOfVariables,
+                  "You are trying to interpolate more variables than this "
+                  "container holds.");
+
+    int dimension = -1;
+    auto const indexArray = std::array<size_t, Dimension>{
+        (++dimension, findIndex(dimension, xin))...};
+
+    return std::array<double, sizeof...(whichVariables)>{
+        (RecursiveInterpolationKernel<
+            decltype(*this), Dimension - 1, whichVariables,
+            CompressedIndexRecursionInitial_t, Xt...>::interpolate(*this,
+                                                                   indexArray,
+                                                                   xin...))...};
+  };
+
+  template <
+      size_t N, typename... Xt,
+      Requires<(std::is_convertible_v<typename std::remove_cv_t<Xt>, double> &&
+                ...)> = nullptr>
+  inline std::array<double, N> interpolate(std::array<size_t, N> whichVariables,
+                                           Xt const&... xin) const {
+    static_assert(
+        sizeof...(Xt) == Dimension,
+        "You need to provide the correct number of interpolation points");
+    static_assert(N <= NumberOfVariables,
+                  "You are trying to interpolate more variables than this "
+                  "container holds.");
+
+    int dimension = -1;
+    auto const indexArray = std::array<size_t, Dimension>{
+        (++dimension, findIndex(dimension, xin))...};
+
+    std::array<double, N> interpolatedValues{};
+
+    for (int i = 0; i < N; ++i)
+      interpolatedValues[i] = RecursiveInterpolationKernelGeneral<
+          decltype(*this), Dimension - 1, CompressedIndexRecursionInitial_t,
+          Xt...>::interpolate(*this, indexArray, whichVariables[i], xin...);
+
+    return interpolatedValues;
+  };
+
+  template <size_t N, size_t... I>
+  inline std::array<double, N> interpolate_helper(
+      std::array<size_t, N> whichVariables, std::array<double, Dimension>& xin,
+      std::index_sequence<I...>) const {
+    return interpolate(whichVariables, xin[I]...);
+  };
+
+  template <size_t N, typename Indices = std::make_index_sequence<Dimension>>
```

Comment:
remove `Indices` and move it down to `interpolator_helper(which_variables, x, std::make_index_sequence<Dimension>{})`